### ISSUE TITLE
Inherit gambit game in Sage

### ISSUE THREAD

**Issue by drvinceknight** [theref/sage-game-theory#31]
- [ ] Initialisation:
  - [ ] ~~Not pass anything and construct game like in docs for gambit api (but without having to pain through the `int` necessity). This makes no assumption with regards to number of players.~~
  - [x] Pass a single matrix so that a zero sum game is formed. This assumes N=2.
  - [x] Pass two matrices (N=2)
  - [ ] Pass something like a bi-matrix...

**Comment by drvinceknight** [comment]
This seems tricky.

**Comment by theref** [comment]
instead of a bi-matrix just leave it as 2 matrices, far clearer for the user

i'm not sure how we're going to get around the `int()` problem without basically rewriting the `gambit.Game()` class. This strikes me ass something that would need to be fixed at API level.

**Comment by drvinceknight** [comment]
I don't think that is needed. You can simply in the init method convert the
inputs surely?

**Comment by drvinceknight** [comment]
Also 'for clarity of user' depends on the user. Someone who has been taught
in terms of bi matrices would find that easier. Easy enough to have a bit
matrix class and/or just grab lists. In init you can just test for them.
On 12 Jun 2014 18:13, vincent.knight@gmail.com wrote:

I don't think that is needed. You can simply in the init method convert the
inputs surely?
On 12 Jun 2014 17:56, "James Campbell" notifications@github.com wrote:

instead of a bi-matrix just leave it as 2 matrices, far clearer for the user

i'm not sure how we're going to get around the int() problem without
basically rewriting the gambit.Game() class. This strikes me ass something
that would need to be fixed at API level.

**Comment by drvinceknight** [comment]
I had misunderstood re integer things.

We will figure out a way around it but not a high priority.

**Comment by drvinceknight** [comment]
This is nice (tab completion: clearly shows that inheritance is working):

![screenshot 2014-06-14 11 10 18](https://cloud.githubusercontent.com/assets/2131546/3278183/5bbdfc3c-f3ac-11e3-9a54-d1c48e84d1bd.png)

I think we can close this issue (raising an enhancement issue for the int thing and the bi-matrix thing which should be doable). What do you think?

**Comment by theref** [comment]
Yeah definitely
